docs/release-management.md

```diff
@@ -0,0 +1,159 @@
+that need to be carefully executed in order across multiple projects.
+
+This process can be automated as described in the following sections.
+
+## What's a release
+
+   A submariner release is the sum of all components, tagged on a specific set of versions,
+   that work and are tested together.
+   
+   The components include:
+   
+   * subctl
+   * submariner-operator
+   * submariner core (submariner engine, route agent, globalnet ipam)
+   * lighthouse (lighthouse agent, lighthouse coredns)
+   * shipyard
+   * admiral
+   
+  
+## A release repository
+
+   A release repository (submariner-io/releases) can be used to track and automate
+   the whole release process.
+   
+   Yaml files in this repository can be used to control the definition of a release, 
+   
+   ```
+      releases/0.5-<name>/0.5.0-<name>.yaml
+      releases/0.5-<name>/0.5.1-bugfix-for....yaml
+      ...
+   ```
+
+## A release yaml
+
+   A release yaml can have the following format:
+   
+   ```yaml
+    ---
+    version: 0.5.0
+    name: the mighty 0.5 release! # The name will be used for the GH release
+    release-notes: 
+      - bugfixes: >
+        * we fixed this bug
+      - features: >
+        * this feature was added
+        * and this other one
+      - upgrades: >  # optional
+        Upgrade notes important to anybody upgrading from a previous version
+      - deprecations: > # optional
+        Features or settings that have been deprecated and need to be updated
+        by the administrator.
+
+    components:
+      - submariner: <commit-id> 
+      - lighthouse: <commit-id>
+      - submariner-operator: <commit-id>
+      - submariner-charts: <commit-id>
+      - shipyard: <commit-id>
+      - admiral: <commit-id>
+
+   ```
+
+## verification of the release yaml
+
+The verification step would:
+* Verify the format of the yaml file, it should be valid yaml and have the required
+  fields to execute the release.
+
+* Verify that passed CI runs exist for the commit-ids proposed for the components,
+  we may have another yaml file in the releases repo defining the jobs that we consider
+  stable, as sometimes we have less stable jobs because of new features.
+
+* Verify that submariner-operator dependencies to projects in go.mod are pointing
+  to the proposed submariner, lighthouse, and shipyard releases, because this affects
+  the behaviour of `subctl verify` to be up-to-date with the e2e tests of those repositories.
+
+* Verify that projects have Dockerfile.dapper pinned to the specific shipyard release.
+
+* Verify that submariner/lighthouse are using the specific admiral commit ID, otherwise stop.
+
+## merge of the release yaml
+
+* A script will run in Github actions, executing step by step the tagging and verification
+of the multiple components.
+
+It will retry builds if necessary.
+
+* Shipyard & Admiral will be tagged for reference
```

Comment:
I don't think this would make sense, as we're stating that we're expecting them to be "ready" already before the release.

i.e.
```
Verify that submariner/lighthouse are using the specific admiral commit ID, otherwise stop.
```
If we tag admiral, dependabot will sync it automatically otherwise someone will have to manually sync.

```
Verify that projects have Dockerfile.dapper pinned to the specific shipyard release.
```
Unless you want them pinned to an arbitrary image version (e.g. `v0.5.0-8-gdf0d54`) beforehand, it would be simpler to have them point to a more coherent tag.

Perhaps release can be a 2 phase approach, one would be a "pre-release" stage upon which these are tagged with -preX (X would be some number that we determine either manually or auto incremented), and then automation could pin everything to that version.

Then the actual "release" where everything else that's in this doc make sense.

We could add this in the future of course, doesnt have to happen now, but this seems like a more coherent way of doing it unless you want to have still some manual steps involved in the process (besides the writing of the release yaml of course).

Reply:
I have tried dependabot in the past and didn't work, may be because I didn't use a "final" release and was 0.5.0-pre0, may be we should start releasing 0.x.0 0.x.1 0.x.2 for admiral by the end of a sprint.